core/src/main/java/org/apache/iceberg/io/ResolvingFileIO.java

```diff
@@ -219,7 +230,7 @@ DelegateFileIO io(String location) {
 
   @VisibleForTesting
   String implFromLocation(String location) {
-    return SCHEME_TO_FILE_IO.getOrDefault(scheme(location), FALLBACK_IMPL);
+    return schemeToFileIo.getOrDefault(scheme(location), FALLBACK_IMPL);
```

Comment:
If FileIO is not initalized but this API is called on empty fileIO, we used to look up from static `SCHEME_PROPERTY_PREFIX`, now it will be an empty map if initialize is not called.

Reply:
`schemeToFileIo.clear()` part was ok. I think this should use `DEFAULT_SCHEME_TO_FILE_IO`, if `schemeToFileIo` is empty.

Reply:
I changed the approach by populating the map at creation time.